pkg/codegen/docs/gen.go

```diff
@@ -1757,11 +1757,40 @@ func (mod *modContext) gen(fs codegen.Fs) error {
 	dctx := mod.docGenContext
 	modName := mod.getModuleFileName()
 
-	addFile := func(name, contents string) {
+	seen := map[string]struct{}{}
+
+	addFile := func(name, contents string, prefixes []string) {
+		var noConflict bool
+		for _, prefix := range prefixes {
+			candidate := prefix + name
+			if _, exists := seen[candidate]; exists {
+				continue
+			}
+			name = candidate
+			noConflict = true
+			seen[name] = struct{}{}
+			break
+		}
+
+		if !noConflict {
+			glog.Error("skipping unresolvable duplicate file name: ", name)
+			return
+		}
+		seen[name] = struct{}{}
 		p := path.Join(modName, name, "_index.md")
 		fs.Add(p, []byte(contents))
 	}
 
+	// Generate the index files.
+	idxData := mod.genIndex()
+	buffer := &bytes.Buffer{}
+	err := dctx.templates.ExecuteTemplate(buffer, "index.tmpl", idxData)
+	if err != nil {
+		return err
+	}
+
+	addFile("", buffer.String(), []string{"", "mod-"})
```

Comment:
Given the check for seen by name (here "") I can't see how this would ever register a conflict, but the bug report raised for this was explicitly about a conflicting module and resource. Feels like we've missed something to cover that.